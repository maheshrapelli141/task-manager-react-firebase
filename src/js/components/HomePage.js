import React from 'react';
import TaskManagement from '../../TaskManagement.jpg';

const HomePage = () => (
  <img src={TaskManagement} alt="Why task management?" width="100%" />
);

export default HomePage;
