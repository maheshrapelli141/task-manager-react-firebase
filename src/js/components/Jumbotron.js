import React from 'react';
import PropTypes from 'prop-types';

const Jumbotron = ({ children }) => (
  <div className="jumbotron jumbotron-fluid">
    <div className="container">{children}</div>
  </div>
);

Jumbotron.propTypes = {
  children: PropTypes.node.isRequired
};

export default Jumbotron;
