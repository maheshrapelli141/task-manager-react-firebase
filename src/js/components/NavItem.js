import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

const NavItem = ({ path, label }) => (
  <li className="nav-item">
    <NavLink to={path} activeClassName="active" className="nav-link">
      {label}
    </NavLink>
  </li>
);

NavItem.propTypes = {
  path: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired
};

export default NavItem;
