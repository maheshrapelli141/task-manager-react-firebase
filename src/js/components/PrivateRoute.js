import React from 'react';
import { connect } from 'react-redux';
import { Route, Redirect } from 'react-router-dom';

const PrivateRoute = ({ component: Component, ...rest }) => (
  <Route 
    {...rest} 
    render={props => (
      rest.user
        ? <Component {...props} />
        : <Redirect to={{ pathname: '/login', state: { from: props.location } }} />
      )
    } 
  />
)

const mapStateToProps = state => {
  const { user } = state;
  return {
    user
  };
}

export default connect(mapStateToProps)(PrivateRoute);