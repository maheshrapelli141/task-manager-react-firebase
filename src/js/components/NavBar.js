import React from 'react';
import { connect } from 'react-redux';

import NavItem from './NavItem';
import { auth, provider } from '../util/firebase';

const login = () => { auth.signInWithPopup(provider) }
const logout = () => { auth.signOut() }

const NavBar = props => (
  <nav className="navbar navbar-expand-lg navbar-dark bg-primary">
    <a className="navbar-brand" href="#">
      Navbar
    </a>
    <button
      className="navbar-toggler"
      type="button"
      data-toggle="collapse"
      data-target="#navbarContent"
    >
      <span className="navbar-toggler-icon" />
    </button>

    <div className="collapse navbar-collapse" id="navbarContent">
      {props.user
        ?
          <ul className="navbar-nav mr-auto">
            <NavItem path="/task" label="Task" />
            <NavItem path="/report" label="Report" />
            <button type="button" className="btn btn-primary btn-right" onClick={logout}  >Logout</button>
          </ul>
        :
          <ul className="navbar-nav mr-auto">
            <button type="button" className="btn btn-primary btn-right" onClick={login}  >Login</button>
          </ul>
      }

    </div>
  </nav>
);

const mapStateToProps = state => {
  const { user } = state;
  return {
    user
  };
}

export default connect(mapStateToProps)(NavBar);
