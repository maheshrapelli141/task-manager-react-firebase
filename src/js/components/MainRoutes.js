import React from 'react';
import { Switch, Route, Redirect } from 'react-router-dom';

import HomePage from './HomePage';
import TaskPage from './TaskPage';
import ReportPage from './ReportPage';
import PrivateRoute from './PrivateRoute';

const MainRoutes = () => (
  <Switch>
    <Route exact path="/" component={HomePage} />
    <PrivateRoute path="/task" component={TaskPage} />
    <PrivateRoute path="/report" component={ReportPage} />
    <Redirect to="/" />
  </Switch>
);

export default MainRoutes;
