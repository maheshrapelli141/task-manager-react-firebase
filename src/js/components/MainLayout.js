import React from 'react';
import { HashRouter } from 'react-router-dom';

import NavBar from './NavBar';
import MainRoutes from './MainRoutes';

const MainLayout = () => (
  <HashRouter hashType="noslash">
    <div className="wrapper">
      <NavBar />
      <main>
        <div className="card">
          <div className="card-body">
            <MainRoutes />
          </div>
        </div>
      </main>
    </div>
  </HashRouter>
);

export default MainLayout;
