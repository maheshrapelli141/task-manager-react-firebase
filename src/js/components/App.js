import React from 'react';
import { connect } from 'react-redux';
import { auth } from '../util/firebase';
import MainLayout from './MainLayout';
import { login } from '../actions/user.actions';

class App extends React.Component {
  constructor(props) {
    super(props);
    
  }

  componentDidMount() {
    auth.onAuthStateChanged((user) => {
      this.props.dispatch(login(user)) 
    });
  }


  render() {
    return (
      <div id="mainContainer">
        <MainLayout />
      </div>);
  }
}
const mapStateToProps = state => {
  const { user } = state;
  return {
    user
  }
};
export default connect(mapStateToProps)(App);
