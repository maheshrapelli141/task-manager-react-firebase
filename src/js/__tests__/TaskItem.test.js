import React from 'react';
import { mount } from 'enzyme';

import { TaskItem } from '../components/TaskItem';
import { addDays, toDateString } from '../util/DateUtil';

const now = '2018-01-01';
Date.now = jest.fn().mockReturnValue(now);

const setup = customProps => {
  const props = {
    id: '1',
    name: 'some task',
    category: 'a category',
    description: 'lorem ipsum',
    createdDate: now,
    reminderDate: now,
    dueDate: toDateString(addDays(now, 8)),
    resolved: false,
    dispatch: jest.fn(),
    ...customProps
  };

  const wrapper = mount(<TaskItem {...props} />);

  return {
    props,
    wrapper
  };
};

describe('TaskItem component', () => {
  afterEach(() => {
    jest.restoreAllMocks();
  });

  it('should render correctly', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render card with border-warning when task is due in 3 days', () => {
    const { wrapper } = setup({ dueDate: toDateString(addDays(now, 3)) });
    expect(wrapper.find('.card.border-warning')).toHaveLength(1);
  });

  it('should render card with border-danger when task is over due', () => {
    const { wrapper } = setup({ dueDate: toDateString(addDays(now, -1)) });
    expect(wrapper.find('.card.border-danger')).toHaveLength(1);
  });

  it('should render card with border-success when task is resolved', () => {
    const { wrapper } = setup({ resolved: true });
    expect(wrapper.find('.card.border-success')).toHaveLength(1);
  });

  it('shoud call dispatch when button is clicked', () => {
    const { wrapper, props } = setup();
    expect(props.dispatch.mock.calls).toHaveLength(0);
    wrapper.find('button').simulate('click');
    expect(props.dispatch.mock.calls).toHaveLength(1);
  });
});
