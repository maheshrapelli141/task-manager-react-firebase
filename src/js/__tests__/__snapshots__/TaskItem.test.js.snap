// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskItem component should render correctly 1`] = `
<TaskItem
  category="a category"
  createdDate="2018-01-01"
  description="lorem ipsum"
  dispatch={[MockFunction]}
  dueDate="2018-01-09"
  id="1"
  name="some task"
  reminderDate="2018-01-01"
  resolved={false}
>
  <div
    className="col-12 col-md-4"
  >
    <div
      className="card"
    >
      <div
        className="card-header"
      >
        a category
      </div>
      <div
        className="card-body"
      >
        <h5
          className="card-title"
        >
          some task
        </h5>
        <h6
          className="card-subtitle mb-2 text-muted"
        >
          Due on 
          2018-01-09
        </h6>
        <p
          className="card-text"
        >
          lorem ipsum
        </p>
      </div>
      <div
        className="card-footer text-center"
      >
        <button
          className="btn btn-block btn-primary"
          onClick={[Function]}
        >
          Resolve
        </button>
        <small
          className="text-muted"
        >
          Created on 
          2018-01-01
        </small>
      </div>
    </div>
  </div>
</TaskItem>
`;
